python/adelphi/adelphi/nb.py

```diff
@@ -6,10 +6,30 @@
 from adelphi.exceptions import KeyspaceSelectionException, TableSelectionException, ExportException
 from adelphi.export import BaseExporter
 
-SEQS={}
-SEQS["text"] = "Mod(1000000000); ToString() -> String"
-DISTS={}
-DISTS["text"] = "Hash(); Mod(1000000000); ToString() -> String"
+MAX_NUMERIC_VAL = 1000 ** 3
+RAMPUP_SCENARIO = "run driver=cql tags=phase:rampup cycles={} threads=auto"
+MAIN_SCENARIO = "run driver=cql tags=phase:main cycles={} threads=auto"
+
+CQL_TYPES={}
+CQL_TYPES["text"] = "Mod({}); ToString() -> String"
+CQL_TYPES["ascii"] = "Mod({}); ToString() -> String"
+CQL_TYPES["int"] = "Mod({}); ToInt() -> int"
+CQL_TYPES["tinyint"] = "Mod({}); ToByte() -> java.lang.Byte"
+CQL_TYPES["smallint"] = "Mod({}); ToShort() -> java.lang.Short"
+CQL_TYPES["bigint"] = "Mod({}); ToLong() -> long"
+CQL_TYPES["float"] = "Mod({}); ToFloat() -> java.lang.Float"
+CQL_TYPES["double"] = "Mod({}); ToDouble() -> double"
+CQL_TYPES["decimal"] = "ModuloToBigDecimal({}) -> java.math.BigDecimal"
+CQL_TYPES["boolean"] = "Mod({}); ToBoolean() -> java.lang.Boolean"
+CQL_TYPES["varchar"] = "Mod({}); ToString() -> String"
+CQL_TYPES["varint"] = "ModuloToBigInt({}) -> java.math.BigInteger"
+CQL_TYPES["blob"] = "Mod({}); ToByteBuffer() -> java.nio.ByteBuffer"
+CQL_TYPES["timestamp"] = "Mod({}); ToDate() -> java.util.Date"
+CQL_TYPES["date"] = "Mod({}); LongToLocalDateDays() -> com.datastax.driver.core.LocalDate"
+CQL_TYPES["time"] = "Mod({}); ToLong() -> long"
+CQL_TYPES["uuid"] = "Mod({}); ToHashedUUID() -> java.util.UUID"
+CQL_TYPES["timeuuid"] = "Mod({}); ToTimeUUIDMax() -> java.util.UUID"
+CQL_TYPES["inet"] = "Mod({}); ToInetAddress() -> java.net.InetAddress"
```

Comment:
This has been discussed elsewhere but it seemed worth adding a note here as well since this could easily be a place somebody might land while doing some investigation.

Note that we are deliberately not including support for the "duration" type introduced in Cassandra 4.0.  Support for generating sequences of this type wasn't built into nosqlbench at the time we started work.  The functionality has since been added (see https://github.com/nosqlbench/nosqlbench/releases/tag/nosqlbench-4.15.33) but since it wasn't ubiquitous at this writing we were left with a few options:

* Include support for the type and specify that users who wish to run Adelphi against schemas with duration types for primary keys must use nosqlbench 4.15.33 or greater
* Leave the type unsupported for now (with the understanding it will be added in later)

The second option was judged to be less problematic than the first, although it's certainly possible this could be revisited as we gather additional data.